akka-http-core/src/test/scala/akka/http/impl/engine/parsing/ResponseParserSpec.scala

```diff
@@ -85,6 +86,17 @@ abstract class ResponseParserSpec(mode: String, newLine: String) extends FreeSpe
         closeAfterResponseCompletion shouldEqual Seq(false)
       }
 
+      "a response with an unrecognized status code" in new Test {
+        // A client must understand the class of any status code, as indicated by the first digit, and
+        // treat an unrecognized status code as being equivalent to the x00 status code of that class
+        // https://tools.ietf.org/html/rfc7231#section-6
+        """HTTP/1.1 456 Totally unrecognized
+          |Content-Length: 0
+          |
+          |""" should parseTo(HttpResponse(TotallyUnrecognized))
```

Comment:
So it creates a custom status code 'on demand' when it encounters an unknown status code. Seems reasonable! (though not obviously the same as "treat an unrecognized status code as being equivalent to the x00 status code of that class")

Reply:
I agree, the spec is not 100% clear on what "treat an unrecognized status code as being equivalent to the x00 status code of that class" means. But in Akka HTTP context it seems reasonable to think that this means that an unrecognized status code should have the same `isSuccess`, `isFailure`, `isRedirection` values as other statuses from its class. And that it should not prevent further processing of the response.

Reply:
Agreed!

Reply:
Are there any actions planned for this PR? One of the production projects in my company would benefit if these changes are approved and end up in Akka HTTP release.

Reply:
> x00 status code of that class

I interpret that as that status codes should be rounded to hundreds if they are not known. But I think this approach here is fine as well -- after all, it's only the parser, so you still need to interpret it.